src/flow_solver/flow_solver_cases/periodic_entropy_tests.cpp

```diff
@@ -0,0 +1,304 @@
+#include "periodic_entropy_tests.h"
+#include "physics/physics_factory.h"
+
+namespace PHiLiP {
+
+namespace FlowSolver {
+
+//=========================================================
+// TEST FOR ENTROPY CONSERVATION/STABILITY ON PERIODIC DOMAINS (EULER/NS)
+//=========================================================
+
+template <int dim, int nstate>
+PeriodicEntropyTests<dim, nstate>::PeriodicEntropyTests(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : PeriodicCubeFlow<dim, nstate>(parameters_input)
+        , unsteady_data_table_filename_with_extension(this->all_param.flow_solver_param.unsteady_data_table_filename+".txt")
+{
+    this->euler_physics = std::dynamic_pointer_cast<Physics::Euler<dim,dim+2,double>>(
+            PHiLiP::Physics::PhysicsFactory<dim,nstate,double>::create_Physics(&(this->all_param)));
+}
+
+template <int dim, int nstate>
+double PeriodicEntropyTests<dim,nstate>::get_constant_time_step(std::shared_ptr<DGBase<dim,double>> dg) const
+{
+
+    using FlowCaseEnum = Parameters::FlowSolverParam::FlowCaseType;
+    const FlowCaseEnum flow_case = this->all_param.flow_solver_param.flow_case_type;
+    const double CFL = this->all_param.flow_solver_param.courant_friedrichs_lewy_number;
+    // For Euler simulations, use CFL
+    const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+    const double approximate_grid_spacing = (this->domain_right-this->domain_left)/pow(number_of_degrees_of_freedom_per_state,(1.0/dim));
+
+    double constant_time_step=0;
+    if (flow_case == FlowCaseEnum::isentropic_vortex){
+        // Using dt = CFL * delta_x/U_infinity, consistent with Ranocha's choice (Relaxation Runge Kutta methods... 2020)
+        // U_infinity is initialized as M_infinity
+        constant_time_step = CFL * approximate_grid_spacing / this->all_param.euler_param.mach_inf;
+    } else if (flow_case == FlowCaseEnum::kelvin_helmholtz_instability){
+        /*
+        const double max_wave_speed = this->compute_integrated_quantities(*dg, IntegratedQuantityEnum::max_wave_speed);
+        constant_time_step = CFL * approximate_grid_spacing / max_wave_speed;
+        */
+        // TEMP using same as is defined in periodic turbulence for consistency with some existing results
+        const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+        const double approximate_grid_spacing = (this->domain_right-this->domain_left)/pow(number_of_degrees_of_freedom_per_state,(1.0/dim));
```

Comment:
These variables have already been defined above (and are redefined again inside `if`).

Reply:
Fixed, thanks for pointing that out.